pkg/health/heartbeat/heartbeat.go

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package heartbeat
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/gardener/etcd-backup-restore/pkg/errors"
+	"github.com/gardener/etcd-backup-restore/pkg/etcdutil"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/coordination/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// Heartbeat is a struct for the heartbeat package
+type Heartbeat struct {
+	logger         *logrus.Entry
+	heartbeatTimer *time.Timer
+	etcdConfig     *etcdutil.EtcdConnectionConfig
+	leaseDuration  int32
+	K8sClient      client.Client
+}
+
+// NewHeartbeat returns the heartbeat object.
+func NewHeartbeat(hconfig *brtypes.HealthConfig, logger *logrus.Entry, etcdConfig *etcdutil.EtcdConnectionConfig) (*Heartbeat, error) {
+	leaseDuration := int32(hconfig.LeaseDuration.Seconds())
+	if leaseDuration < 1 {
```

Comment:
Is it already part of `pkg/types/health.go`
I had added it here as an extra check initially and later thought it might provide a secondary safety check
wdyt?